recon/src/protocol.rs

```diff
@@ -497,17 +496,26 @@ where
     ) -> Result<RemoteStatus> {
         match message {
             ResponderMessage::InterestResponse(interests) => {
-                let mut ranges = Vec::with_capacity(interests.len());
-                for interest in interests {
-                    ranges.push(
-                        self.common
-                            .recon
-                            .initial_range(interest)
-                            .await
-                            .context("querying initial range")?,
-                    );
+                if interests.is_empty() {
+                    to_writer
+                        .send(ToWriter::Finish)
+                        .await
+                        .map_err(|err| anyhow!("{err}"))
+                        .context("sending finish")?;
+                    return Ok(RemoteStatus::Finished);
```

Comment:
do we want to do an early return on the other exit step too? [that is, here](https://github.com/ceramicnetwork/rust-ceramic/pull/367/files#diff-7c57e2a3f12a04af8303d7883cea6c88b855ecea5ecb0356653b3d4a2b4945f8R539)

Reply:
no we do not want any early returns from this